core/src/main/java/org/apache/calcite/schema/Table.java

```diff
@@ -82,4 +84,23 @@ public interface Table {
    */
   boolean rolledUpColumnValidInsideAgg(String column, SqlCall call,
       @Nullable SqlNode parent, @Nullable CalciteConnectionConfig config);
+
+  /**
+   * Gets Looker Explore metadata if available.
+   */
+  default HashMap<String, Object> getTableMetadata() {
+    HashMap<String, Object> map = new HashMap<>();
```

Comment:
Let's avoid hash maps in APIs whenever possible, even internal ones. This is how helltool ended up with so many functions with a single hash map argument instead of individual arguments.

The best thing for here is probably to define a new data class called `ExtraTableMetadata` that functions like the hash map, but with all the expected fields spelled out.